flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphSchedulingTest.java

```diff
@@ -1,637 +0,0 @@
-
-		executionGraph.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
-		executionGraph.scheduleForExecution();
-
-		final TestingLogicalSlot slot = createTestingSlot();
-		final CompletableFuture<?> releaseFuture = slot.getReleaseFuture();
-		slotFuture1.complete(slot);
-
-		// cancel should change the state of all executions to CANCELLED
-		executionGraph.cancel();
-
-		// complete the now CANCELLED execution --> this should cause a failure
-		slotFuture2.complete(new TestingLogicalSlotBuilder().createTestingLogicalSlot());
-
-		Thread.sleep(1L);
-		// release the first slot to finish the cancellation
-		releaseFuture.complete(null);
-
-		// NOTE: This test will only occasionally fail without the fix since there is
-		// a race between the releaseFuture and the slotFuture2
-		assertThat(executionGraph.getTerminationFuture().get(), is(JobStatus.CANCELED));
-	}
-
-	/**
-	 * Tests that a partially completed eager scheduling operation fails if a
-	 * completed slot is released. See FLINK-9099.
-	 */
-	@Test
-	public void testSlotReleasingFailsSchedulingOperation() throws Exception {
-		final int parallelism = 2;
-
-		final JobVertex jobVertex = new JobVertex("Testing job vertex");
-		jobVertex.setInvokableClass(NoOpInvokable.class);
-		jobVertex.setParallelism(parallelism);
-		final JobGraph jobGraph = new JobGraph(jobVertex);
-		jobGraph.setScheduleMode(ScheduleMode.EAGER);
-
-		final ProgrammedSlotProvider slotProvider = new ProgrammedSlotProvider(parallelism);
-
-		final LogicalSlot slot = createSingleLogicalSlot(new DummySlotOwner(), new SimpleAckingTaskManagerGateway(), new SlotRequestId());
-		slotProvider.addSlot(jobVertex.getID(), 0, CompletableFuture.completedFuture(slot));
-
-		final CompletableFuture<LogicalSlot> slotFuture = new CompletableFuture<>();
-		slotProvider.addSlot(jobVertex.getID(), 1, slotFuture);
-
-		final ExecutionGraph executionGraph = createExecutionGraph(jobGraph, slotProvider);
-
-		executionGraph.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
-		executionGraph.scheduleForExecution();
-
-		assertThat(executionGraph.getState(), is(JobStatus.RUNNING));
-
-		final ExecutionJobVertex executionJobVertex = executionGraph.getJobVertex(jobVertex.getID());
-		final ExecutionVertex[] taskVertices = executionJobVertex.getTaskVertices();
-		assertThat(taskVertices[0].getExecutionState(), is(ExecutionState.SCHEDULED));
-		assertThat(taskVertices[1].getExecutionState(), is(ExecutionState.SCHEDULED));
-
-		// fail the single allocated slot --> this should fail the scheduling operation
-		slot.releaseSlot(new FlinkException("Test failure"));
-
-		assertThat(executionGraph.getTerminationFuture().get(), is(JobStatus.FAILED));
-	}
-
-	/**
-	 * Tests that all slots are being returned to the {@link SlotOwner} if the
-	 * {@link ExecutionGraph} is being cancelled. See FLINK-9908
-	 */
-	@Test
-	public void testCancellationOfIncompleteScheduling() throws Exception {
```

Comment:
How does the `DefaultScheduler` cancels pending slot requests if the user calls `DefaultScheduler.cancel()`?

Reply:
I think that on invocation of `DefaultScheduler.cancel()`, the job will be fully canceled and the JobMaster will shutdown and `close` the SlotPool. `SlotPoolImpl#close()` will release all slots, both allocated and available ones. So looks to me we the scheduler does not need to take care of slot releasing. Please correct me if I missed anything.

Reply:
added `SlotPoolImpl#testShutdownCancelsAllPendingRequests `

Reply:
I think it is not strictly needed by we are relying here on the behaviour of some other components. In general it would be nice if a component is self-contained. I think we don't have to add it now. We just should be aware of it.